### PR TITLE
Refactor lib API and update dependencies

### PR DESCRIPTION
**Pull request by GerardPaligot**
* Expose only final Composable and the ViewModel
* Use `remember` instead of `Application` class for the state
* Update dependency (including Compose)
* Migrate to Gradle Version Catalog
* Update README file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,33 @@ An Android client for Open-Feeedback https://github.com/HugoGresse/open-feedback
 
 The SDK exposes a regular View and a Composable if you want to try [compose](https://developer.android.com/jetpack/compose). Please consult the [sample-app](sample-app/src/main/java/io/openfeedback/android/sample/MainActivity.kt) for usage.
```
Comment (martinbonnin) [inline README.md]: We might want to change that now 😄

File: build-logic/build.gradle.kts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 plugins {
-    id("org.jetbrains.kotlin.jvm").version("1.6.10")
-    id("java-gradle-plugin")
+    `embedded-kotlin`
```
Comment (martinbonnin) [inline build-logic/build.gradle.kts:2]: 👍

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
```
Comment (martinbonnin) [inline gradle/wrapper/gradle-wrapper.properties:6]: 👍

File: README.md (hunk 1 above)
Comment (GerardPaligot) [reply README.md]: Oh yes. Maybe I can just remove it and add at the end of the Usage section, something like:
If you want to see an example, please check the sample app (+ link to the module)

**Review by martinbonnin — APPROVED** [review]
Very cool 👍 . As we move to this being more public, what do you think of renaming `SessionFeedbackContainer` -> `OpenFeedbackLayout` ? Or something else? Maybe just `SessionFeedback` ?

File: README.md (hunk 1 above)
Comment (martinbonnin) [reply README.md]: I as mainly mentioning the regular `View` thing that's not there anymore. Sample app can stay here or moved lower, no strong opinion there

**Comment by GerardPaligot** [comment]
I though about it during the refactoring, so totally agree. I can suggest you:
- `SessionDetailsContainer` > `OpenFeedback` (and rename the current `OpenFeedback` to `OpenFeedbackState`)
- `SessionFeedback` > `OpenFeedbackLayout`

**Comment by martinbonnin** [comment]
> SessionDetailsContainer > OpenFeedback

Works for me 👍 .

> and rename the current OpenFeedback to OpenFeedbackState

 `OpenFeedbackState` feels a bit weird to me for something that's more like a config to me but I don't have much Compose XP so no strong opinion there. Works for me too 👍

**Comment by GerardPaligot** [comment]
> OpenFeedbackState feels a bit weird to me for something that's more like a config to me but I don't have much Compose XP so no strong opinion there. Works for me too 👍

You're right! I renamed it `OpenFeedbackConfig`.

File: README.md (hunk 1 above)
Comment (GerardPaligot) [reply README.md]: Done

**Comment by GerardPaligot** [comment]
Thanks @martinbonnin!

**Comment by martinbonnin** [comment]
Thank you! That's a lot of nice improvements 🤩 .

**Pull request merged**